plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -395,6 +482,50 @@ private ParquetFileWriter createParquetFileWriter(Location path, List<DeltaLakeC
         }
     }
 
+    private boolean isNoRowsRetained(String path, Roaring64NavigableMap pastDeletionVectors, FileDeletion deletion)
+    {
+        LongBitmapDataProvider rowsDeletedByDelete = deletion.rowsDeletedByDelete();
+        LongBitmapDataProvider rowsDeletedByUpdate = deletion.rowsDeletedByUpdate();
+        int retainedCount = 0;
+        try (ConnectorPageSource connectorPageSource = createParquetPageSource(Location.of(path)).get()) {
+            long filePosition = 0;
+            while (!connectorPageSource.isFinished()) {
+                Page page = connectorPageSource.getNextPage();
+                if (page == null) {
+                    continue;
+                }
+
+                int positionCount = page.getPositionCount();
+                for (int position = 0; position < positionCount; position++) {
+                    if (!pastDeletionVectors.contains(filePosition) && !rowsDeletedByDelete.contains(filePosition) && !rowsDeletedByUpdate.contains(filePosition)) {
+                        retainedCount++;
+                    }
+                    filePosition++;
+                }
```

Comment:
Since we're dealing with bitmaps, there should be a shorter way to tackle this.
We can OR the pastDeletionVectors, rowsDeletedByDelete and rowsDeletedByUpdate bitvectors up front to require look-up in only one bitmap in the loop.
Even better way would be to create a bitmap out of a range using `new RoaringBitmap64().addRange(start, end)`, intersect that with the above OR'ed bitmap and then check `isEmpty` on the result.

Reply:
Regarding the 2nd suggestion, do the `start` & `end` arguments come from `Roaring64Bitmap#first` and `Roaring64Bitmap#last` of the combined bitvectors? e.g. 
```java
        Roaring64Bitmap deletedRows = new Roaring64Bitmap();
        deletedRows.or(pastDeletionVectors);
        deletedRows.or(deletion.rowsDeletedByUpdate());
        deletedRows.or(deletion.rowsDeletedByUpdate());

        Roaring64Bitmap ranges = new Roaring64Bitmap();
        ranges.addRange(deletedRows.first(), deletedRows.last());
...
```
(The `first()` method throws `Empty HighLowContainer`)

Reply:
I meant something like this
```
Roaring64Bitmap filePositions = new Roaring64Bitmap();
filePositions.addRange(0, rowCount);
filePositions.andNot(deletedRows);
return filePositions.isEmpty();
```